test/unit/puppetlabs/puppetserver/ruby/http_client_test.clj

```diff
@@ -191,6 +200,33 @@ jruby-config :- jruby-schemas/JRubyConfig
           (is (= "401" (.runScriptlet sc "$response.code")))
           (is (= "access denied" (.runScriptlet sc "$response.body"))))))))
 
+(deftest http-compressed-requests
+  (jetty9/with-test-webserver ring-app-decompressing-gzipped-request port
+    (with-scripting-container sc
+      (with-http-client sc port {:use-ssl false}
+        (testing "GZIP compression format"
+          (let [compress "{ :compress => :gzip }"
+                body "howdy"]
+            (.runScriptlet sc (format "$response = $c.post('/', '%s', {}, %s)"
+                                      body
+                                      compress))
+            (is (= "200" (.runScriptlet sc "$response.code")))
+            (is (= body (.runScriptlet sc "$response.body")))))
+
+       (testing "invalid compression format"
+         (let [compress "{ :compress => :bunk }"]
+           (is (= "Unsupported compression specified for request: bunk"
+                  (.runScriptlet
+                   sc
+                   (str "begin;"
+                        "  $response = $c.post('/', 'foo', {}, "
+                        compress
+                        ");"
+                        "  'No error raised from post';"
+                        "rescue ArgumentError => e;"
+                        "  e.message;"
+                        "end"))))))))))
```

Comment:
Yeah, I had something earlier that tried to let the exception / error bubble up from JRuby into the Java layer but that created a pretty gross Wolfe Principle violation that I couldn't figure out how to get rid of easily. JRuby wants to log a full stack trace for any uncaught error that bubbles up. This is definitely gross but seemed a bit less gross.